src/redist/targets/packaging/windows/clisdk/registrykeys.wxs

```diff
@@ -34,5 +34,47 @@
         </RegistryKey>
       </Component>
     </ComponentGroup>
+
+    <!-- Set values for DOTNETHOME if not passed in.  
+         These searches prefer the existing registry key and fallback to probing the default location if registry key isn't present -->
+    <Property Id="DOTNETHOME_X86_SEARCH">
+      <DirectorySearch Id="dirDOTNETHOME_X86_search" Path="[ProgramFilesFolder]dotnet" AssignToProperty="yes">
+        <FileSearch Id="fileDOTNETHOME_X86_search" Name="dotnet.exe" />
+      </DirectorySearch>
+      <RegistrySearch Id="regDOTNETHOME_X86_search" Root="HKLM"  Key="SOFTWARE\dotnet\Setup\InstalledVersions\x86" Name="InstallLocation" Win64="no" Type="raw" />
+    </Property>
+    <Property Id="DOTNETHOME_X64_SEARCH">
+      <DirectorySearch Id="dirDOTNETHOME_X64_search" Path="[ProgramFiles64Folder]dotnet" AssignToProperty="yes">
+        <FileSearch Id="fileDOTNETHOME_X64_search" Name="dotnet.exe" />
+      </DirectorySearch>
+      <RegistrySearch Id="regDOTNETHOME_X64_search" Root="HKLM"  Key="SOFTWARE\dotnet\Setup\InstalledVersions\x64" Name="InstallLocation" Win64="no" Type="raw" />
+    </Property>
+    <Property Id="DOTNETHOME_X64_ALT_SEARCH">
+      <DirectorySearch Id="dirDOTNETHOME_X64_ALT_search" Path="[ProgramFiles64Folder]dotnet\x64" AssignToProperty="yes">
+        <FileSearch Id="fileDOTNETHOME_X64_ALT_search" Name="dotnet.exe" />
+      </DirectorySearch>
+      <RegistrySearch Id="regDOTNETHOME_X64_ALT_search" Root="HKLM"  Key="SOFTWARE\dotnet\Setup\InstalledVersions\x64" Name="InstallLocation" Win64="no" Type="raw" />
+    </Property>
+    <Property Id="DOTNETHOME_ARM64_SEARCH">
+      <DirectorySearch Id="dirDOTNETHOME_ARM64_search" Path="[ProgramFiles64Folder]dotnet" AssignToProperty="yes">
+        <FileSearch Id="fileDOTNETHOME_ARM64_search" Name="dotnet.exe" />
+      </DirectorySearch>
+      <RegistrySearch Id="regDOTNETHOME_ARM64_search" Root="HKLM"  Key="SOFTWARE\dotnet\Setup\InstalledVersions\arm64" Name="InstallLocation" Win64="no" Type="raw" />
+    </Property>
+
+    <!-- Set the DOTNETHOME property if not passed in and we found a path -->
+    <SetProperty Id="DOTNETHOME_X86" Value="[DOTNETHOME_X86_SEARCH]" Before="CostFinalize">
```

Comment:
I think I got this. The only time we'd be in trouble is a clean install, where the bundle fails to pass DOTNETHOME_xxx and the search won't find anything because we've never been installed and that would technically point to an authoring error. I need to check how the SDK is authored into VS because there VS takes the place of the bundle to pass in DOTNET_HOME to the MSI

Reply:
We shouldn't have to pass in defaults.  A clean install won't detect it as it runs searches up front in the bundle.  We can run similar searches here so that we detect it after the host was installed.  The bundle's searches miss this because they run before the host is installed.

In general we have to be very careful when using public properties to control installed content: think through all scenarios of  install/repair/patch(if supported)/uninstall/upgrade to make sure property values are correct.  Never count on the MSI being "called correctly" to get that stuff right.

>  I need to check how the SDK is authored into VS because there VS takes the place of the bundle to pass in DOTNET_HOME to the MSI

We should remove those.  The meaning of `DOTNETHOME` public properties is "final install location of dotnet root directory".  This assumption is made in SDK bundle authoring and one that is a public contract with users that we cannot break.  We're changing the default for x64.  If someone is specifying that (including VS bundle) by duplicating the defaults it needs to change to reflect the new default.